Rollout the latest LLVM revision for the safer C++ bots.

rniwa opened the pull request:
#### 5d256bf1ee6680ff338e284fd233b3171157913d
<pre>
Rollout the latest LLVM revision for the safer C++ bots.
<a href="https://bugs.webkit.org/show_bug.cgi?id=304103">https://bugs.webkit.org/show_bug.cgi?id=304103</a>

Reviewed by Geoffrey Garen.

This rollout includes:
 - Recognizing a getter on a dependent smart pointer type as a safe
 - Treat variable captures of a lambda passed to WTF::ScopeExit and WTF::makeVisitor as safe
 - Fix an assertion failure in coroutine body in webkit.UncountedLambdaCapturesChecker.
 - Support recognizing an assignment to an +1 out argument as not a leak
 - Treating a weak Objective-C variable as safe
 - Recognizing [allocObj() init] pattern as safe in alpha.webkit.UnretainedCallArgsChecker.
 - Support for WTF::move in place of std::move
 - Ignoring variable captures in a lambda which gets immediately called
 - Ignoring a variable declaration in &quot;if&quot; statement if the then statements are trivial

* Tools/CISupport/safer-cpp-llvm-version:

Canonical link: <a href="https://commits.webkit.org/304435@main">https://commits.webkit.org/304435@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff5b0c53b2864269344562f6a1a3762bc9c9b1b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135478 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/7856 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/46763 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143172 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87162 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d8b3c07c-76de-4f00-9366-33d239a363e0) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137347 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7703 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103528 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70884 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/79da480a-cec0-4d34-b927-91dd5b46de36) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138424 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6097 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121432 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84395 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/014e19ea-72fc-4d05-a39e-c09a9a43a911) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5874 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3481 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/3782 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115075 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39618 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/145923 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7541 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40189 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111895 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7580 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6328 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112267 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5726 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117734 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61436 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20895 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7593 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/35850 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7340 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7559 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7441 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren reviewed (approved):
Would be nice to test removing the SUPPRESS macros Chris added recently.

rniwa committed:
Rollout the latest LLVM revision for the safer C++ bots.
https://bugs.webkit.org/show_bug.cgi?id=304103

Reviewed by Geoffrey Garen.

This rollout includes:
 - Recognizing a getter on a dependent smart pointer type as a safe
 - Treat variable captures of a lambda passed to WTF::ScopeExit and WTF::makeVisitor as safe
 - Fix an assertion failure in coroutine body in webkit.UncountedLambdaCapturesChecker.
 - Support recognizing an assignment to an +1 out argument as not a leak
 - Treating a weak Objective-C variable as safe
 - Recognizing [allocObj() init] pattern as safe in alpha.webkit.UnretainedCallArgsChecker.
 - Support for WTF::move in place of std::move
 - Ignoring variable captures in a lambda which gets immediately called
 - Ignoring a variable declaration in "if" statement if the then statements are trivial

* Tools/CISupport/safer-cpp-llvm-version:

Canonical link: https://commits.webkit.org/304435@main
